### ISSUE TITLE
"make debug" fails: undefined reference to `cryptonote::BlockchainDB::get_block_from_height(unsigned long const&) const'

### ISSUE THREAD

**Issue by danrmiller** [monero-project/monero#1792]
"Make debug" fails on several platforms.

`
[ 95%] Linking CXX executable ../../bin/monero-utils-deserialize
../cryptonote_basic/libcryptonote_basic.so: undefined reference to `cryptonote::BlockchainDB::get_block_from_height(unsigned long const&) const'
collect2: error: ld returned 1 exit status
src/blockchain_utilities/CMakeFiles/cn_deserialize.dir/build.make:116: recipe for target 'bin/monero-utils-deserialize' failed`

For example: 
https://build.getmonero.org/builders/monero-tests-ubuntu-16.04-i686/builds/2/steps/compile/logs/stdio

**Comment by moneromooo-monero** [comment]
https://github.com/monero-project/monero/pull/1794 should fix it

**Comment by Jaqueeee** [comment]
@moneromooo-monero Still having similar issue with debug build on osx.
```
Scanning dependencies of target cryptonote_basic
[ 69%] Linking CXX shared library libcryptonote_basic.dylib
Undefined symbols for architecture x86_64:
  "cryptonote::BlockchainDB::get_block_from_height(unsigned long long const&) const", referenced from:
      cryptonote::HardFork::get_block_version(unsigned long long) const in hardfork.cpp.o
      cryptonote::HardFork::reorganize_from_block_height(unsigned long long) in hardfork.cpp.o
      cryptonote::HardFork::rescan_from_block_height(unsigned long long) in hardfork.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [src/cryptonote_basic/libcryptonote_basic.dylib] Error 1
make[2]: *** [src/cryptonote_basic/CMakeFiles/cryptonote_basic.dir/all] Error 2
```

**Comment by moneromooo-monero** [comment]
Looks like blockchain_db and cryptonote_basic require each other.

**Comment by tdprime** [comment]
Commit 49efd3add9f7b9bbcbd2526538846f6d8e58ac86 broke things

**Comment by moneromooo-monero** [comment]
Does https://github.com/moneromooo-monero/bitmonero/tree/crossdep fix it ?

**Comment by tdprime** [comment]
This build succeeded.

What is being lost by doing this?

**Comment by moneromooo-monero** [comment]
Nothing. A file is moved from cryptonote_basic to cryptonote_core, and it is the one that was causing a cross dependency. cryptonote_core and cryptonote_basic were the same lib until recently.
Thanks for checking, I'll PR.

**Comment by kenshi84** [comment]
Given Issue #1851, it's surprising to me that PR #1804 solved this issue. @danrmiller @Jaqueeee @tdprime Can you confirm?

I created another [patch](https://github.com/monero-project/monero/files/829486/diff_virtual.txt) for this - could you please try this by doing
```
git apply diff_virtual.txt
```
from the project root on the latest master b67877af6f7192a302453e542c266a5cfc3182a7?

**Comment by danrmiller** [comment]
@kenshi84 I have the same issue as #1804 with https://github.com/monero-project/monero/commit/b67877af6f7192a302453e542c266a5cfc3182a7. 

Your patch works for me where I tried (osx).

**Comment by kenshi84** [comment]
Could you please close this issue which was solved by #1853?
